packages/common-utils/src/renderChartConfig.ts

```diff
@@ -816,36 +816,51 @@ function translateMetricChartConfig(
       whereLanguage: 'sql',
     };
   } else if (metricType === MetricsDataType.Sum && metricName) {
+    const time_bucket_col = '`__hdx_time_bucket2`';
+    const value_high_col = '`__hdx_value_high`';
+    const value_high_prev_col = '`__hdx_value_high_prev`';
+    const time_expr = timeBucketExpr({
+      interval: chartConfig.granularity || 'auto',
+      timestampValueExpression:
+        chartConfig.timestampValueExpression || 'TimeUnix',
+      dateRange: chartConfig.dateRange,
+      alias: '__hdx_time_bucket2',
+    });
     return {
       ...restChartConfig,
       with: [
         {
           name: 'RawSum',
-          sql: chSql`SELECT *,
-               any(Value) OVER (ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING) AS PrevValue,
-               any(AttributesHash) OVER (ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING) AS PrevAttributesHash,
-               IF(AggregationTemporality = 1,
-                  Value,IF(Value - PrevValue < 0 AND AttributesHash = PrevAttributesHash, Value,
-                      IF(AttributesHash != PrevAttributesHash, 0, Value - PrevValue))) as Rate
+          sql: chSql`SELECT ${time_expr}, AttributesHash, last_value(a.Value) AS ${value_high_col},
```

Comment:
My understanding of this query is we compute rates and sum them into accumulative values, and then compute rates again. I wonder if its more efficient to use accumulative values at the beginning (if `AggregationTemporality` is 1, then adding values up).

Reply:
It should be simplified. I had to dig into the aggregation function docs.